src/Controls/src/Core/Shell/Shell.cs

```diff
@@ -49,7 +49,18 @@ static void OnBackButonBehaviorPropertyChanged(BindableObject bindable, object o
 
 		/// <summary>Bindable property for attached property <c>NavBarIsVisible</c>.</summary>
 		public static readonly BindableProperty NavBarIsVisibleProperty =
-			BindableProperty.CreateAttached("NavBarIsVisible", typeof(bool), typeof(Shell), true);
+			BindableProperty.CreateAttached("NavBarIsVisible", typeof(bool), typeof(Shell), true, propertyChanged: OnNavBarIsVisibleChanged);
+
+		private static void OnNavBarIsVisibleChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			// Nav bar visibility change is only interesting from the Shell down to the current Page.
+			// Make sure the ShellToolbar knows about any possible change.
+			Shell shell = bindable as Shell
+				?? (bindable as BaseShellItem)?.FindParentOfType<Shell>()
+				?? (bindable as Page)?.FindParentOfType<Shell>();
+
+			shell.OnPropertyChanged(NavBarIsVisibleProperty.PropertyName);
```

Comment:
Should we add some null safety here? Just in case someone tries to apply this attached property to something that's not a page?

Reply:
Good catch